### PR TITLE
feature(locksmith): Retry transactions

### PR DESCRIPTION
**Pull request by julien51**
# Description

Retrying sending transactions up to 3 times with exponential backoff.

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by julien51** [comment]
Rebase https://github.com/unlock-protocol/unlock/pull/11263 first

**Comment by julien51** [comment]
cc @searchableguy ready for review please :)

**Comment by searchableguy** [comment]
Worried this might result in timeout.

**Comment by julien51** [comment]
> Worried this might result in request timeout. I think trying once more is fine instead of thrice.

Ok I will change to one retry (2 tries total).

> We will need to use nounce manager eventually (ethers has one in v5 as experimental but stable in v6)

Agreed.

File: locksmith/src/fulfillment/retries.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+const MAX_RETRIES = 2
```
Comment (julien51) [inline locksmith/src/fulfillment/retries.ts:1]:
```suggestion
const MAX_RETRIES = 1
```

**Comment by julien51** [comment]
@searchableguy PTAL :)

**Review by searchableguy — APPROVED** [review]

**Pull request merged**